### PR TITLE
Serialize ActivityLog entries to string properly

### LINKED ISSUE #4521: Error at "Review History" click after changing "Notes for Reviewers" for a version
Steps to reproduce:
1. Go to manage version page for any submitted add-on i.e.  https://addons-dev.allizom.org/en-US/developers/addon/add-on-wx/versions/1530325
2. Add or change "Notes for Reviewers"
3. Go to "Manage Status & Versions" page and click "Review History" link for that version

Expected results:
There are no display or layout issues. The "Review History" link work as expected.

Actual results:
An error is displayed while "Review History" link is clicked for a version with changed reviewer notes.

Notes/Issues:
- issue is reproducing for listed (awaiting or approved) add-on, unlisted add-on, mixed versions add-on, etc.
- there are no issues if "Notes for Reviewers" are added during submission.
Verified on FF51(Win 7). Issue is reproducing all around AMO-servers.
Video for this issue: 
long-issue

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #4521.

### CODE REVIEW

**Comment by wagnerand** [comment]
We don't need the approval notes in the review history, since they are already part of that version. Some developers include a lot of information there, so we want to avoid duplicating so much content.
Why can't we go without a comment?

**Comment by eviljeff** [comment]
Without the comments it's a fairly useless update - you have to delve into devhub to find out what they are.  Plus we don't have a record of what they changed to, when they're next changed.

**Comment by wagnerand** [comment]
You don't have to go to devhub. The approval notes are always the first part of the review history for each version.
Earlier versions of the approval notes are not really relevant, all that counts is the latest version.

**Comment by eviljeff** [comment]
You're only thinking about the reviewer tools.  ActivityMail exposes the update in devhub versions page and by email.  If you added the event to set the information requested flag, why do we send out an email and show the event at all?

**Comment by wagnerand** [comment]
We send out emails to reviewers to notify them that the review is no longer blocked and can be continued.

**Comment by eviljeff** [comment]
So an email is sent saying nothing other than to check the website to see what the current state of the approval notes are.  There might be something you'd want to reply to the developer about but you won't get the chance to see until you go the review page.  The same logic could be applied to all of the email updates we send out for developer replies - why include the text when the place with all the information is the reviewer tools so just log in there.

Either way, we need to get this bug fixed so developers can access the review history again and redesign-by-pull-request-comment is rarely a good idea so @jvillalobos what do you want the comment to be when the developer changes the approval notes?    I think it's consistent to include everything the developer wrote; @wagnerand wants a terse comment just saying they've changed.  

It will be shown in the review history in versions&files in devhub; shown in the reviewer tools in the version history; emailed to everyone involved in the review/addon except the user who changed them.

**Comment by wagnerand** [comment]
> So an email is sent saying nothing other than to check the website to see what the current state of the approval notes are. There might be something you'd want to reply to the developer about but you won't get the chance to see until you go the review page. The same logic could be applied to all of the email updates we send out for developer replies - why include the text when the place with all the information is the reviewer tools so just log in there.

That is a fair point. I didn't include the comment originally since there is no extra value of having it in the email. The reviewer needs to log in to AMO anyway to continue the review.

> Either way, we need to get this bug fixed so developers can access the review history again and redesign-by-pull-request-comment is rarely a good idea so @jvillalobos what do you want the comment to be when the developer changes the approval notes? I think it's consistent to include everything the developer wrote; @wagnerand wants a terse comment just saying they've changed.

I am not opposed to include the comment, but for the record want to emphasize two things
1) This pull request itself changes the design, not my comment.
2) This pull request works around the technical issue that some events need to have a comment and others don't. Without some kind of check or enforcement (either way) in the code, this will break again once someone adds another event, which will likely happen in the near future.

> It will be shown in the review history in versions&files in devhub; shown in the reviewer tools in the version history; emailed to everyone involved in the review/addon except the user who changed them.

I am not sure whether sending this event to the other developers of the add-on (if any) is more helpful or more confusing. It could see it go either way.

**Comment by eviljeff** [comment]
Fair points... effectively the core of the problem is that the Activity API, and the emails, didn't envision any events without a comment (because at the time it was written all appropriate events had comments).  So it can either be thought of as a) log events being incorrectly `log()`d without a comment or b) there is a flaw in the design of the Activity API and the email functionality.

Either a) or b) is fixable (though I'd split (b) into a quick fix and another pull because it'll take longer)

**Comment by wagnerand** [comment]
Sgtm!

File: src/olympia/activity/serializers.py [hunk 1]
```diff
@@ -24,7 +24,8 @@ def __init__(self, *args, **kwargs):
         self.to_highlight = kwargs.get('context', []).get('to_highlight', [])
 
     def get_comments(self, obj):
-        comments = obj.details['comments'] if obj.details else obj.to_string()
+        comments = (obj.details['comments'] if obj.details
+                    else obj.to_string().unescape())
```
Comment (diox) [inline src/olympia/activity/serializers.py]: Since finding the right fix was not trivial, can you add a quick comment explaining why we have to do this ?

Comment (eviljeff) [inline src/olympia/activity/serializers.py]: https://github.com/mozilla/addons-server/pull/4598/commits/96dd7aabd0d14a9b9250a10b8c2078a523ae8ef8

**Pull request merged**